### ISSUE TITLE
Incorrect Conversion of Array with Rest Element to Zod Schema

### ISSUE THREAD

**Issue by TheCrowdsecMatt** [fabien0102/ts-to-zod#256]
## Bug description

Incorrect conversion of an array type with a rest element into a Zod schema that includes z.any(), which is not the intended behavior.

### Input

```ts
export interface Resident {
    id: string;
    name: string;
    addresses: [Address, ...Address[]];
}

export interface Address {
  street:string;
  zipCode: number;
};
```

### Expected output

```ts
import { z } from "zod"

export const addressSchema = z.object({
  street: z.string(),
  zipCode: z.number()
})

export const residentSchema = z.object({
  id: z.string(),
  name: z.string(),
  
  // Could also be -> abilitations: z.array(abilitationSchema).min(1);
  addresses: z.array(addressSchema).nonempty() 
})
```

### Actual output

```ts
import { z } from "zod"

export const addressSchema = z.object({
  street: z.string(),
  zipCode: z.number()
})

export const residentSchema = z.object({
  id: z.string(),
  name: z.string(),
  addresses: z.tuple([addressSchema, z.any()])
})

```

**Versions**

- Typescript: `v5.4.5`
- Zod: `v3.22.4`

**Comment by tvillaren** [comment]
Hello @TheCrowdsecMatt and thanks for reporting. 

I see in [Zod document about tuples](https://zod.dev/?id=tuples) that the rest operator is supported so I think the expected output should be 

```ts
export const residentSchema = z.object({
  id: z.string(),
  name: z.string(),
  addresses: z.tuple(addressSchema).rest(addressSchema);
})
```

**Comment by TheCrowdsecMatt** [comment]
Hello @tvillaren 
I agree your suggestion makes sense also 👍 
Thanks for the PR creation.

**Comment by tvillaren** [comment]
Thanks, I released it in 3.10.0 
https://github.com/fabien0102/ts-to-zod/pull/258
